src/main/java/com/fcgl/madrid/search/controller/UserSearchController.java

```diff
@@ -25,16 +25,36 @@ public void setServices(UserSearchService categoryService) {
     /**
      * We want to keep track of the searches a user has made
      * This endpoint simply adds an entry to the UserSearch database table based on the search a user has made
-     *
+     * <p>
      * TODO: Based on what we find out on elastic search. We might have to make an api request to elastic search from here as well.
      */
     @PostMapping(path = "/query")
     public ResponseEntity<Response<String>> addUserSearchQuery(@Valid @RequestBody UserSearchRequest request) {
         return userSearchService.addQueryToTable(request);
     }
 
-    @GetMapping(path = "/query") //TODO returns a 500 not sure why
-    public ResponseEntity<Response<SearchHistoryResponse>> getQueriesForUser(@Valid UserId request) {
-            return userSearchService.searchByUserId(request);
+    /**
+     * This endpoint returns a paginated list of queries that a user has made.
+     *
+     * @param id       specify user id to be queried
+     * @param pageSize Optional parameter: specify number of queries per page
+     * @param page     Optional parameter: specify page number of results
+     * @return list of results
+     */
+    @GetMapping(path = "query") //expect page index starting from 1, so subtract one for querying purposes
+    public ResponseEntity<Response<SearchHistoryResponse>> getRecentQueriesForUser(@Valid UserId id,
```

Comment:
Let's change the param to "SearchHistoryRequest" with the object having Long userId, Integer size, and Integer page?

With @NotNull for `userId` and default values for `page` and `size`?

Reply:
Sure.